test/unit/data/test_galaxy_mapping.py

```diff
@@ -920,6 +920,31 @@ def test_can_manage_private_dataset(self):
         assert security_agent.can_manage_dataset(u_from.all_roles(), d1.dataset)
         assert not security_agent.can_manage_dataset(u_other.all_roles(), d1.dataset)
 
+    def test_history_hid_counter_is_expired_after_next_hid_call(self):
+        u = model.User(email="hid_abuser@example.com", password="password")
+        h = model.History(name="History for hid testing", user=u)
+        self.persist(u, h)
+        state = inspect(h)
+        assert h.hid_counter == 1
+        assert 'hid_counter' not in state.unloaded
+        assert 'id' not in state.unloaded
+
+        h._next_hid()
+
+        assert 'hid_counter' in state.unloaded  # this attribute has been expired
+        assert 'id' not in state.unloaded  # but other attributes have NOT been expired
```

Comment:
Oh, so because we bypass the current session completely this works. That's neat!

Reply:
And if it didn't it would've probably wreaked havoc in the tests, where a lot of HDAs would be committed with a hid.

Reply:
Yep! I also turned on echo=True for the engine when I was working on this to make sure there were no surprises.